app/services/include_params_validator.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class IncludeParamsValidator
+  include ActiveModel::Validations
+
+  attr_reader :relationships, :supported_relationships
+
+  validates_each :relationships, allow_blank: true do |record, _attr, values|
+    unsupported_values = values - record.supported_relationships
+
+    unless unsupported_values.empty?
+      record.errors.add 'Bad request', "#{unsupported_values} is not supported. Valid values are: #{record.supported_relationships}"
+    end
+  end
+
+  def initialize(relationships, supported_relationships)
+    @relationships = relationships
+    @supported_relationships = supported_relationships
+  end
+
+  def fully_validate!
+    raise ValidationError, self if invalid?
+  end
+
+  class ValidationError < StandardError
```

Comment:
I find using this might be simpler, what do you think? 
`ValidationError = Class.new(StandardError)`

Reply:
update: doesn't work